src/API/Wallet/Owner/Handlers/GetNewWalletAddressHandler.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <Core/Config.h>
+#include <Wallet/WalletManager.h>
+#include <Net/Clients/RPC/RPC.h>
+#include <Net/Servers/RPC/RPCMethod.h>
+#include <API/Wallet/Owner/Models/Errors.h>
+#include <optional>
+
+class GetNewWalletAddressHandler : public RPCMethod
+{
+public:
+	GetNewWalletAddressHandler(const IWalletManagerPtr& pWalletManager, const TorProcess::Ptr& pTorProcess)
+		: m_pWalletManager(pWalletManager), m_pTorProcess(pTorProcess) { }
+	virtual ~GetNewWalletAddressHandler() = default;
+
+	RPC::Response Handle(const RPC::Request& request) const final
+	{
+		if (!request.GetParams().has_value())
+		{
+			return request.BuildError(RPC::Errors::PARAMS_MISSING);
+		}
+
+		Json::Value tokenJson = JsonUtil::GetRequiredField(request.GetParams().value(), "session_token");
+		SessionToken token = SessionToken::FromBase64(tokenJson.asString());
+
+		KeyChainPath newPath = m_pWalletManager->IncreaseKeyChainPathIndex(token);
```

Comment:
`IncreaseKeyChainPathIndex` should probably either take in a `parent_path` (which you should supply as "m/0/1"), or it should be renamed to something like `IncreaseAddressKeyChainPathIndex`